### ISSUE TITLE
Sway sends wl_keybaord modifiers event before sending keymap

### ISSUE THREAD

**Issue by tmccombs** [swaywm/sway#8169]
- **Sway Version:**
  - sway version 1.9

- **Description:**

See https://github.com/emersion/slurp/issues/123 and

**Comment by emersion** [comment]
Reading the wlroots source code, I don't understand how this can happen.

**Comment by emersion** [comment]
Hm, I suppose it could happen if `wlr_seat_keyboard_notify_modifiers()` is called before `wlr_seat_set_keyboard()`?

The only unguarded call in Sway is in the IME code. Are you using an IME by any chance?

**Comment by tmccombs** [comment]
I'm using wtype to send keyboard events. Maybe that uses the same API as IME?

**Comment by layercak3** [comment]
wtype uses virtual keyboard, not text-input (IME). Does it still happen in master? I'm unable to get it to crash on my machine.

**Comment by tmccombs** [comment]
Yes, I can still reproduce on master. All I have to do to reproduce is run:

```
$ wtype no && slurp
```

**Comment by layercak3** [comment]
The modifiers event comes from calling `wlr_seat_keyboard_send_modifiers` inside `wlr_seat_keyboard_enter`. Removing the virtual keyboard makes `keyboard_state.keyboard` NULL until something like a key press/release happens to call `wlr_seat_set_keyboard`. In this case, by the time `seat_client_create_keyboard` is called (for handling a client's `wl_seat.get_keyboard`), it's still NULL, which prevents `seat_client_create_keyboard` from sending a keymap to the client. Then by the time sway switches keyboard focus to the surface, `wlr_seat_set_keyboard` has still not been called, so enter/modifiers is sent without a keymap ever being sent.

(If you run `$ wtype no && sleep 0.5 && slurp`, it will most likely not segfault as you will probably release the enter key after the virtual keyboard has been removed and before the execution of `slurp`, which will cause `wlr_seat_set_keyboard` before `slurp`.)

What should we do then? Set an arbitrary keyboard when removing one (if there are still keyboards left)? Have a default keymap ready so that `seat_client_create_keyboard` has a keymap to send to new client wl_keyboards even while `keyboard_state.keyboard` is NULL?
